### PR TITLE
Add `{% ChromeDate %}` shortcode for auto-freshing release dates

### LINKED ISSUE #4663: Implement filter to cast dates from Chrome milestone numbers
In the context of the WebSQL deprecation there was a request to @tomayac to update the date in the deprecation post. This could actually be automated by using the data from the [Chrome Release Schedule](https://chromiumdash.appspot.com/schedule) that is also powering the release widget on the homepage.

To achieve this, the task to sync the external data from the respective API needs to be updated at: [/external/build/lib/chrome-release.js](https://github.com/GoogleChrome/developer.chrome.com/blob/main/external/build/lib/chrome-release.js) to not only output the `beta` and `stable` channel information to external/data/chrome-release.json, like this

```json
{
  "channels": {
    "beta": {
      "mstone": 108,
      "key": "beta",
      "stableDate": "2022-11-29T00:00:00.000Z"
    },
    "stable": {
      "mstone": 107,
      "key": "stable",
      "stableDate": "2022-10-25T00:00:00.000Z"
    }
  }
}
```

But contain an additional key `milestones` like this:

```json
{
  "milestones": {
    "m121": {
      "branch_point": "2023-11-16T00:00:00",
      "earliest_beta": "2023-12-07T00:00:00",
      "earliest_beta_ios": "2023-11-29T00:00:00",
      "feature_freeze": "2023-11-02T00:00:00",
      "final_beta": "2024-01-03T00:00:00",
      "final_beta_cut": "2024-01-02T00:00:00",
      "late_stable_date": "2024-01-11T00:00:00",
      "latest_beta": "2023-12-14T00:00:00",
      "stable_cut": "2024-01-02T00:00:00",
      "stable_date": "2024-01-09T00:00:00",
    }
  },
  "channels": {
    ...
  }
}
```

This information could then be used in a new filter `chromeDate`, which could be used in templates like this:

```md
## Web SQL deprecation and removal steps

- [✅ Done.] Web SQL was deprecated and removed for **third-party contexts** in
  **Chromium 97** ({{ 'm97'|chromeDate:'stable_date' }}).
- [✅ Done.] Web SQL access in **insecure contexts** was deprecated as of
  **Chromium 105** ({{ 'm105'|chromeDate:'stable_date' }}) at which time a warning message was shown in the Chrome DevTools Issue
  panel.
```

`stableDate` could also be considered the default date.

### PR DESCRIPTION
**Pull request by matthiasrohmer**
Resolves #4663. This adds a new shortcode `{% ChromeDate %}` which can be used like this:

```md
- [📍 We are here.] Web SQL access in **insecure contexts** is no longer available as of
  **Chromium&nbsp;110** ({% ChromeDate '110' %}). An **enterprise policy** to keep using the feature is available from
  **Chromium&nbsp;110** ({% ChromeDate '110' %}) to **Chromium&nbsp;111** ({% ChromeDate '111' %}).
```

During rendering `{% ChromeDate '111' %}` would become `March 7, 2023`, respecting the locale.

Caveat: the API used is moving, so it might be necessary to update the `MILESTONES_OFFSET` and `MILESTONES_COUNT` config constants sometime.

### CODE REVIEW

File: site/_filters/chromeDate.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const {milestones} = require('../../external/data/chrome-release.json');
+
```
Comment (devnook) [inline site/_filters/chromeDate.js:18]: we might need a lint exception here

**Review by devnook — APPROVED** [review]

**Review by tomayac — APPROVED** [review]
LGTM (in theory, the staging link https://pr-4757-static-dot-dcc-staging.uc.r.appspot.com/blog/deprecating-web-sql/ didn't work), thanks so much!

**Comment by matthiasrohmer** [comment]
Note: after review this can only be merged while the dcc-external-data has an updated version of chrome-release.json. Otherwise the build will fail. It has for *now*, but the external data job from `main` will run in 20 minutes and overwrite the new version. To update, run

```bash
gh pr checkout 4757
node external/build/chrome-release.js
```

and upload `external/data/chrome-release.json` to dcc-external-data via the Google Cloud Console.

**Review by devnook — APPROVED** [review]

**Comment by matthiasrohmer** [comment]
Forwarding to @rachelandrew for handbook review.

**Review by rachelandrew — APPROVED** [review]
Nice! Thank you.

**Comment by tomayac** [comment]
Thank you indeed! This is a small, but super useful addition! `<3`

**Pull request merged**